support/testing/scenarios/setup.go

```diff
@@ -0,0 +1,150 @@
+	"github.com/filecoin-project/specs-actors/actors/builtin/verifreg"
+	"github.com/filecoin-project/specs-actors/actors/util/adt"
+	"github.com/filecoin-project/specs-actors/support/ipld"
+	actor_testing "github.com/filecoin-project/specs-actors/support/testing"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/specs-actors/actors/runtime"
+	"github.com/ipfs/go-cid"
+	"github.com/stretchr/testify/require"
+	cbg "github.com/whyrusleeping/cbor-gen"
+)
+
+func NewVMWithSingletons(ctx context.Context, t *testing.T, additionalAccounts int) (*VM, []address.Address) {
+	store := ipld.NewADTStore(ctx)
+
+	lookup := map[cid.Cid]exported.BuiltinActor{}
+	for _, ba := range exported.BuiltinActors() {
+		lookup[ba.Code()] = ba
+	}
+
+	vm := NewVM(ctx, lookup, store)
+
+	rootVerifier := verifregRootAddresses(t)
+
+	emptyMapCID, err := adt.MakeEmptyMap(vm.store).Root()
+	require.NoError(t, err)
+	emptyArrayCID, err := adt.MakeEmptyArray(vm.store).Root()
+	require.NoError(t, err)
+	emptyMultimapCID, err := adt.MakeEmptyMultimap(vm.store).Root()
+	require.NoError(t, err)
+
+	FIL := big.NewInt(1e18)
+
+	initializeActor(ctx, t, vm, &system.State{}, builtin.SystemActorCodeID, builtin.SystemActorAddr, big.Zero())
+
+	addrTreeRoot, accountAddrs := initAddressTree(t, vm, rootVerifier, additionalAccounts)
+	initState := initactor.ConstructState(addrTreeRoot, "scenarios")
+	initState.NextID = abi.ActorID(builtin.FirstNonSingletonActorId + additionalAccounts)
+	initializeActor(ctx, t, vm, initState, builtin.InitActorCodeID, builtin.InitActorAddr, big.Zero())
+
+	rewardState := reward.ConstructState(big.Mul(big.NewInt(100), FIL))
+	initializeActor(ctx, t, vm, rewardState, builtin.RewardActorCodeID, builtin.RewardActorAddr, big.Max(big.NewInt(14e8), FIL))
+
+	cronState := cron.ConstructState(cron.BuiltInEntries())
+	initializeActor(ctx, t, vm, cronState, builtin.CronActorCodeID, builtin.CronActorAddr, big.Zero())
+
+	powerState := power.ConstructState(emptyMapCID, emptyMultimapCID)
+	initializeActor(ctx, t, vm, powerState, builtin.StoragePowerActorCodeID, builtin.StoragePowerActorAddr, big.Zero())
+
+	marketState := market.ConstructState(emptyMapCID, emptyArrayCID, emptyMultimapCID)
+	initializeActor(ctx, t, vm, marketState, builtin.StorageMarketActorCodeID, builtin.StorageMarketActorAddr, big.Zero())
+
+	vrState := verifreg.ConstructState(emptyMapCID, rootVerifier.idAddr)
+	initializeActor(ctx, t, vm, vrState, builtin.VerifiedRegistryActorCodeID, builtin.VerifiedRegistryActorAddr, big.Zero())
+
+	// burnt funds
+	initializeActor(ctx, t, vm, &account.State{Address: builtin.BurntFundsActorAddr}, builtin.AccountActorCodeID, builtin.BurntFundsActorAddr, big.Zero())
+
+	// verified root
+	rootVerifierState := &account.State{Address: rootVerifier.pubAddr}
+	initializeActor(ctx, t, vm, rootVerifierState, builtin.AccountActorCodeID, rootVerifier.idAddr, big.Zero())
+
+	// additional accounts
+	initialAccountBalance := big.Mul(big.NewInt(10_000), FIL)
```

Comment:
Please make this a parameter. I'd rather separate the creation of accounts like this from the singletons. A few ways this could go:
- pass a slice of AccountSpec objects to this method, each spec having a pubkey and initial balance
- don't init accounts here at all, but export a new method for adding an account to state directly

The accounts should come after the other actors, so create the init and other singletons first, and _then_ add the accounts (if any)